### ISSUE TITLE
etsdemo unable to show Python file on PySide2 from PyPI

### ISSUE THREAD

**Issue by kitchoi** [enthought/traitsui#1452]
On master (not released), if the etsdemo application is run with PySide2 5.15 from PyPI, a file cannot be shown and there is an error traceback:

```
Traceback (most recent call last):
  File "/Users/kchoi/Work/ETS/py39-venv/etsdemo-env/lib/python3.9/site-packages/traitsui/qt4/tree_editor.py", line 943, in _on_tree_sel_changed
    ui = object.edit_traits(
  File "/Users/kchoi/Work/ETS/py39-venv/etsdemo-env/lib/python3.9/site-packages/traits/has_traits.py", line 1670, in edit_traits
    return view.ui(
  File "/Users/kchoi/Work/ETS/py39-venv/etsdemo-env/lib/python3.9/site-packages/traitsui/view.py", line 462, in ui
    ui.ui(parent, kind)
  File "/Users/kchoi/Work/ETS/py39-venv/etsdemo-env/lib/python3.9/site-packages/traitsui/ui.py", line 244, in ui
    self.rebuild(self, parent)
  File "/Users/kchoi/Work/ETS/py39-venv/etsdemo-env/lib/python3.9/site-packages/traitsui/qt4/toolkit.py", line 147, in ui_subpanel
    ui_panel.ui_subpanel(ui, parent)
  File "/Users/kchoi/Work/ETS/py39-venv/etsdemo-env/lib/python3.9/site-packages/traitsui/qt4/ui_panel.py", line 62, in ui_subpanel
    _ui_panel_for(ui, parent, True)
  File "/Users/kchoi/Work/ETS/py39-venv/etsdemo-env/lib/python3.9/site-packages/traitsui/qt4/ui_panel.py", line 68, in _ui_panel_for
    ui.control = control = _Panel(ui, parent, is_subpanel).control
  File "/Users/kchoi/Work/ETS/py39-venv/etsdemo-env/lib/python3.9/site-packages/traitsui/qt4/ui_panel.py", line 130, in __init__
    self.control = panel(ui)
  File "/Users/kchoi/Work/ETS/py39-venv/etsdemo-env/lib/python3.9/site-packages/traitsui/qt4/ui_panel.py", line 259, in panel
    panel = _GroupPanel(content[0], ui).control
  File "/Users/kchoi/Work/ETS/py39-venv/etsdemo-env/lib/python3.9/site-packages/traitsui/qt4/ui_panel.py", line 555, in __init__
    self._add_splitter_items(content, splitter)
  File "/Users/kchoi/Work/ETS/py39-venv/etsdemo-env/lib/python3.9/site-packages/traitsui/qt4/ui_panel.py", line 650, in _add_splitter_items
    panel = self._add_items([item])
  File "/Users/kchoi/Work/ETS/py39-venv/etsdemo-env/lib/python3.9/site-packages/traitsui/qt4/ui_panel.py", line 891, in _add_items
    editor.prepare(inner)
  File "/Users/kchoi/Work/ETS/py39-venv/etsdemo-env/lib/python3.9/site-packages/traitsui/editor.py", line 236, in prepare
    self.init(parent)
  File "/Users/kchoi/Work/ETS/py39-venv/etsdemo-env/lib/python3.9/site-packages/traitsui/qt4/html_editor.py", line 59, in init
    page.setLinkDelegationPolicy(QtWebKit.QWebPage.DelegateAllLinks)
AttributeError: 'PySide2.QtWebEngineWidgets.QWebEnginePage' object has no attribute 'setLinkDelegationPolicy'
```

![Screenshot 2020-12-16 at 16 55 56](https://user-images.githubusercontent.com/3673984/102480568-5c9c9f00-4058-11eb-9079-fd3a0ae85ac2.png)

To reproduce, simply install `etsdemo` from master with PySide2 5.15 from PyPI and launch a Python file in the application.

The cause is #1448 with #1446.  This issue serves to track the behaviour in the etsdemo application with the given PySide2 backend. While this is tied to #1448, this crippling behaviour may block a release of etsdemo if #1448 is not fixed and released. 

Note also that the etsdemo is an application distributed separately from the TraitsUI distribution.

Package versions:
<details>
    <summary> pip list </summary>

```
Package             Version       Location
------------------- ------------- ---------------------------------------
configobj           5.0.6
docutils            0.16
etsdemo             0.2.0.dev1711 /Users/kchoi/Work/ETS/traitsui/ets-demo
importlib-metadata  3.3.0
importlib-resources 3.3.0
pip                 20.2.3
pyface              7.2.0
Pygments            2.7.3
PySide2             5.15.2
setuptools          49.2.1
shiboken2           5.15.2
six                 1.15.0
traits              6.1.1
traitsui            7.1.1
zipp                3.4.0
```

</details>

**Comment by kitchoi** [comment]
Reopening:  etsdemo is a separate distribution. If etsdemo is released from master before the fix to #1448 is released from TraitsUI, then this buggy behaviour will be released as well. To close this issue, there are two options (1) one needs to release TraitsUI with the fix #1448 first, or (2) etsdemo can be released earlier but the flag needs to be reverted in that case before the release.

This issue is open separately so that we can track this bug independently of #1448.

**Comment by aaronayres35** [comment]
I am no longer seeing he error reported (I believe this is expected as #1448 has been closed)
I created a fresh edm env, shelled into and ran:
```
pip install pyside2
pip install -e .
cd ets-demo
pip install -e .
cd ..
edm install pygments numpy
etsdemo
```
and launch the example given in the description.  It looks correct initially.  However, if I then click a different example, such as the "LED display" I see a black screen in the middle section of the app and closing the app gives long traceback
<details>

```
$ etsdemo
2021-04-08 15:59:21.709 Python[476:20161921] ApplePersistenceIgnoreState: Existing state will not be touched. New state will be written to (null)
Traceback (most recent call last):
  File "/Users/aayres/Desktop/traitsui/traitsui/qt4/tree_editor.py", line 933, in _on_tree_sel_changed
    self._clear_editor()
  File "/Users/aayres/Desktop/traitsui/traitsui/qt4/tree_editor.py", line 775, in _clear_editor
    editor._node_ui.dispose()
  File "/Users/aayres/Desktop/traitsui/traitsui/ui.py", line 255, in dispose
    self.finish()
  File "/Users/aayres/Desktop/traitsui/traitsui/ui.py", line 277, in finish
    self.reset(destroy=False)
  File "/Users/aayres/Desktop/traitsui/traitsui/ui.py", line 314, in reset
    editor.dispose()
  File "/Users/aayres/Desktop/traitsui/traitsui/editors/shell_editor.py", line 136, in dispose
    self.update_object, "command_executed", remove=True, dispatch="ui"
  File "/Users/aayres/.edm/envs/dumby/lib/python3.6/site-packages/traits/has_traits.py", line 2330, in observe
    dispatcher=_ObserverDispatchers[dispatch],
  File "/Users/aayres/.edm/envs/dumby/lib/python3.6/site-packages/traits/observation/observe.py", line 59, in observe
    remove=remove,
  File "/Users/aayres/.edm/envs/dumby/lib/python3.6/site-packages/traits/observation/_observe.py", line 54, in add_or_remove_notifiers
    callable_()
  File "/Users/aayres/.edm/envs/dumby/lib/python3.6/site-packages/traits/observation/_observe.py", line 94, in __call__
    step()
  File "/Users/aayres/.edm/envs/dumby/lib/python3.6/site-packages/traits/observation/_observe.py", line 118, in _add_or_remove_extra_graphs
    remove=self.remove,
  File "/Users/aayres/.edm/envs/dumby/lib/python3.6/site-packages/traits/observation/_observe.py", line 54, in add_or_remove_notifiers
    callable_()
  File "/Users/aayres/.edm/envs/dumby/lib/python3.6/site-packages/traits/observation/_observe.py", line 94, in __call__
    step()
  File "/Users/aayres/.edm/envs/dumby/lib/python3.6/site-packages/traits/observation/_observe.py", line 150, in _add_or_remove_maintainers
    change_notifier.remove_from(observable)
  File "/Users/aayres/.edm/envs/dumby/lib/python3.6/site-packages/traits/observation/_observer_change_notifier.py", line 123, in remove_from
    raise NotifierNotFound("Notifier not found.")
traits.observation._exceptions.NotifierNotFound: Notifier not found.
Traceback (most recent call last):
  File "/Users/aayres/Desktop/traitsui/traitsui/qt4/ui_live.py", line 252, in _on_finished
    self.close(accept)
  File "/Users/aayres/Desktop/traitsui/traitsui/qt4/ui_live.py", line 240, in close
    super(_LiveWindow, self).close(rc)
  File "/Users/aayres/Desktop/traitsui/traitsui/qt4/ui_base.py", line 291, in close
    self.ui.dispose(rc)
  File "/Users/aayres/Desktop/traitsui/traitsui/ui.py", line 255, in dispose
    self.finish()
  File "/Users/aayres/Desktop/traitsui/traitsui/ui.py", line 277, in finish
    self.reset(destroy=False)
  File "/Users/aayres/Desktop/traitsui/traitsui/ui.py", line 314, in reset
    editor.dispose()
  File "/Users/aayres/Desktop/traitsui/traitsui/qt4/tree_editor.py", line 257, in dispose
    self._delete_node(self._tree.invisibleRootItem())
  File "/Users/aayres/Desktop/traitsui/traitsui/qt4/tree_editor.py", line 428, in _delete_node
    self._delete_node(cnid)
  File "/Users/aayres/Desktop/traitsui/traitsui/qt4/tree_editor.py", line 428, in _delete_node
    self._delete_node(cnid)
  File "/Users/aayres/Desktop/traitsui/traitsui/qt4/tree_editor.py", line 428, in _delete_node
    self._delete_node(cnid)
  File "/Users/aayres/Desktop/traitsui/traitsui/qt4/tree_editor.py", line 465, in _delete_node
    self._clear_editor()
  File "/Users/aayres/Desktop/traitsui/traitsui/qt4/tree_editor.py", line 775, in _clear_editor
    editor._node_ui.dispose()
  File "/Users/aayres/Desktop/traitsui/traitsui/ui.py", line 252, in dispose
    self.save_prefs()
  File "/Users/aayres/Desktop/traitsui/traitsui/ui.py", line 543, in save_prefs
    toolkit().save_window(self)
  File "/Users/aayres/Desktop/traitsui/traitsui/qt4/toolkit.py", line 338, in save_window
    helper.save_window(ui)
  File "/Users/aayres/Desktop/traitsui/traitsui/qt4/helper.py", line 146, in save_window
    geom = ui.control.geometry()
RuntimeError: Internal C++ object (_GroupSplitter) already deleted.
```

</details>

Possibly this issue can be closed and a separate issue opened

**Comment by aaronayres35** [comment]
I repeated the process I described https://github.com/enthought/traitsui/issues/1452#issuecomment-816194816 and I no longer see the follow up issue I mentioned.  Note that traitsui 7.2 was released which included the fix of #1448.  I think it is safe to close this issue now, so I am doing so.  If we need to reopen we can
